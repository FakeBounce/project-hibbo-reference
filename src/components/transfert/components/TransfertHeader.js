import React from 'react';
import PropTypes from 'prop-types';
import { View, Animated } from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';

import underscore from 'assets/transfert/underscore.png';
import underscoreBlack from 'assets/transfert/underscoreBlack.png';
import underscoreMoneyPotPic from 'assets/transfert/underscoreMoneyPot.png';
import { getTranslations } from 'utils/i18n';
import TransfertHeaderButton from './TransfertHeaderButton';

import {
  STransferHeaderUnderline,
  STransferHeaderImg,
  STransferHeaderUnderLineImg,
} from '../styles';

const styles = EStyleSheet.create({
  container: {
    flex: 0,
    width: '100% - 75',
    flexDirection: 'row',
    marginTop: 10,
    alignSelf: 'center',
    justifyContent: 'space-between',
    marginLeft: 5,
  },
  containerMoneyPot: {
    backgroundColor: 'transparent',
  },
});

const assets = {
  underscore: {
    notSelected: underscore,
    selected: underscoreBlack,
    underscoreMoneyPot: underscoreMoneyPotPic,
  },
};

const TransfertHeader = ({
  isMoneyPot,
  selectedTab,
  callback,
  tabs,
  moveBar,
}) => {
  let source = assets.underscore.notSelected;
  if (isMoneyPot) {
    source = assets.underscore.underscoreMoneyPot;
  }
  return (
    <View>
      <STransferHeaderImg source={source} />
      <View style={[styles.container, isMoneyPot && styles.containerMoneyPot]}>
        {tabs.map((value, key) => {
          return (
            <TransfertHeaderButton
              key={value}
              value={value}
              index={key}
              selectedTab={selectedTab}
              callback={callback}
            />
          );
        })}
      </View>
      <STransferHeaderUnderline
        style={{ transform: [{ translateX: moveBar }] }}
      >
        <STransferHeaderUnderLineImg source={assets.underscore.selected} />
      </STransferHeaderUnderline>
    </View>
  );
};

TransfertHeader.propTypes = {
  selectedTab: PropTypes.number.isRequired,
  callback: PropTypes.func.isRequired,
  tabs: PropTypes.arrayOf(PropTypes.string),
  isMoneyPot: PropTypes.bool.isRequired,
  moveBar: PropTypes.instanceOf(Animated.Value).isRequired,
};

TransfertHeader.defaultProps = {
  tabs: [
    getTranslations('transfert.header.pay'),
    getTranslations('transfert.header.debit'),
    getTranslations('transfert.header.moneyPot'),
  ],
};

export default TransfertHeader;
