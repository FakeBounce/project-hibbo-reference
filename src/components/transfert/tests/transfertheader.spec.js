import React from 'react';
import renderer from 'react-test-renderer';
import {
  Animated,
} from 'react-native';

import TransfertHeader from '../components/TransfertHeader';

describe('TransfertHeader', () => {
  it('should render the component', () => {
    const number = renderer.create(
      <TransfertHeader
        selectedTab={0}
        callback={jest.fn}
        isMoneyPot={false}
        moveBar={new Animated.Value(0)}
      />,
    ).toJSON();

    expect(number).toMatchSnapshot();
  });
});
