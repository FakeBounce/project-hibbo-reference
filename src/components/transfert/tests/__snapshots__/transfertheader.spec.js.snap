// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransfertHeader should render the component 1`] = `
<View>
  <Image
    source={1}
    style={undefined}
  />
  <View
    style={
      Array [
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            undefined,
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        >
          Pay
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            undefined,
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        >
          Debit
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            undefined,
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        >
          Money Pot
        </Text>
      </View>
    </View>
  </View>
  <View
    collapsable={undefined}
    style={
      Object {
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <Image
      source={1}
      style={undefined}
    />
  </View>
</View>
`;
